rero_ils/modules/documents/dojson/contrib/jsontomarc21/model.py

```diff
@@ -582,53 +697,91 @@ def reverse_physical_description(self, key, value):
     return result or None
 
 
+@to_marc21.over('6XX', '^subjects')
+@utils.reverse_for_each_value
+@utils.ignore_value
+def reverse_subjects(self, key, value):
+    """Reverse - subjects.
+
+    Sujet Personne > 600
+    Sujet Organisation > 610 OU 611 Conference
+    Sujet Concept > 650
+    """
+    if entity := value.get('entity'):
+        tag = None
+        query = EntitiesSearch() \
+            .filter('term', pid=entity.get('pid', '???'))
+        if query.count():
+            entity_type = next(query.source('type').scan()).type
+        else:
+            entity_type = entity.get('type') or entity.get('bf:Agent')
+        if entity_type in [EntityType.PERSON, EntityType.ORGANISATION]:
+            result, entity_type, surname, conference = do_contribution(
+                contribution={'entity': entity},
+                source_order=to_marc21.source_order
+            )
+            if entity_type == EntityType.PERSON:
+                tag = '6001_' if surname else '6000_'
+            elif entity_type == EntityType.ORGANISATION:
+                tag = '611__' if conference else '610__'
+        elif entity_type == EntityType.TOPIC:
+            result = do_concept(
+                entity=entity,
+                source_order=to_marc21.source_order
+            )
+            tag = '650__'
+        elif entity_type == EntityType.WORK:
+            if authorized_access_point := entity.get(
+                f'authorized_access_point_{to_marc21.language}'
+            ) or entity.get('authorized_access_point'):
```

Comment:
```python
if access_point := entity.get(f'authorized_access_point_{to_marc21.language}', entity.get('authorized_access_point')):
```
You can merge conditions from parent `elif` and this `if` statement

Reply:
Yes, but in the future we have to change it. I added:
`# TODO: to change in the future if $ref's are used.`